### ISSUE TITLE
Prefer no DRM when multiple options available

### ISSUE THREAD

**Issue by leonardr** [NYPL-Simplified/Simplified-iOS#915]
Overdrive sometimes offers a book in both DRM and no-DRM format. NYPL doesn't have a lot of these but other libraries do. After resolving some issues with early return, I'm confident recommending that clients always fulfill using a no-DRM delivery mechanism when one is available. We shouldn't be encouraging the use of DRM in any way, and using non-DRM fulfillments whenever possible will marginally reduce the risk of DRM SNAFUs.

**Comment by winniequinn** [comment]
@leonardr This is a tricky thing to get right, I think. For example, if a book is available without DRM via an `open-access` link and with DRM via a `borrow` link, the better user experience is actually to use the latter because books checked out via `borrow` links sync. Any thoughts?

**Comment by leonardr** [comment]
I agree with your reasoning and I agree that for SimplyE purposes it's better to follow a DRMed `borrow` link than an `open-access` link. I don't think that case will happen in real life, but you can model that fact by making link relation take priority over media type when making your decision.

I'm talking here about the scenario where you're deciding between two links with the same link relation (`borrow` or `acquisition`). That's the case Overdrive offers.

**Comment by winniequinn** [comment]
@leonardr Understood!

The app currently does not have a notion of acquisition priority: It simply uses the first valid acquisition path it will be able to handle. Prioritizing certain acquisition relations over others (e.g. preferring `borrow` to `open-access`) _and_ prioritizing certain acquisition paths over others based on whether or not a component of a given path will require DRM would take a bit of work.

I can certainly do it, but I wonder if this will be too much complexity once the app is presenting multiple download options. Would the expectation be that the app would coalesce acquisition paths that result in a specific content type into a single user-selectable option before then presenting the set of coalesced options to the user (e.g. "EPUB", "Audiobook")? Would it be better to just present all possible options and details (e.g. "EPUB", "Adobe + EPUB", "Adobe + Audiobook")? Maybe a hybrid where we only show initial components of paths that have a common suffix (e.g. "EPUB", "Adobe + EPUB", "Audiobook")? 

I think I might prefer to leave the app as-is for now and address this when we get to user-selectable content types and can handle all of the implementation cases at once. Does that seem reasonable?

**Comment by leonardr** [comment]
I'm fine with waiting to improve this system if it doesn't follow naturally from your recent work to support bearer token fulfillment.
